metric_tracking/__init__.py

```diff
@@ -0,0 +1,241 @@
+import logging
+import storage
+import mlflow
+
+def do_tracking():
+    available_dates = list(storage.available().keys())
+    available_dates.sort()
+
+    for date_a, date_b in zip(available_dates[:-1], available_dates[1:]):
+        mlflow.start_run()
+        mlflow.log_param("Date A", date_a)
+        mlflow.log_param("Date B", date_b)
+        calculate_score(date_a, date_b)
+        mlflow.end_run()
+
+def calculate_score(date_a, date_b):
+    cluster_a = storage.read(date_a)
+    cluster_b = storage.read(date_b)
+
+    inverse_cluster_a = invert_cluster(cluster_a)
+    inverse_cluster_b = invert_cluster(cluster_b)
+
+    logging.info("calculate score for dates %s - %s" % (date_a, date_b))
+    for cluster_id in inverse_cluster_a.keys():
+        systems_a = frozenset(inverse_cluster_a[cluster_id])
+        systems_b = frozenset(inverse_cluster_b[cluster_id])
+        # So now if you have 100 system ids in first and 160 in second
+        # (100 are the same and we have 60 new), you'll count the score as
+        # 100 / (100+160/2) = 100 / 130 = 0.76.
+        # While it should be 1, if the 100 systems were clustered the same.
+        # We can't say anything about the 60 new systems (unless we do other analysis of them),
+        # so those needs to be ignored.
+        # TODO: only look at systems present in both runs
+
+        number_ids_in_both = len(systems_a.intersection(systems_b))
+
+        median_number_ids_in_cluster = (len(systems_a) + len(systems_b)) / 2.0
+        score = number_ids_in_both / median_number_ids_in_cluster
```

Comment:
Yeah, but then I think that's a deficiency of the clustering code. It should label the same clusters always the same. 
And let's not forget about the purpose of this small PR. It's for laying out the basics of that workflow of improvements :)

Reply:
Its just a consequence of how the k-means algorithm works. One thing to do now to avoid this different initialization of centroids  is to set a random seed like, `KMeans(n_clusters=4, random_state=42)`. That way they will always start at the same place.

You're right, its not a major issue at this stage. But it should generate higher stability scores.